src/reducers/index.ts

```diff
@@ -107,80 +139,73 @@ export default compose(storeFreeze, storeLogger(), combineReducers)({
  * first select the books state then we pass the state to the book
  * reducer's getBooks selector, finally returning an observable
  * of search results.
+ * 
+ * Share memoizes the selector functions and published the result. This means
+ * every time you call the selector, you will get back the same result
+ * observable. Each subscription to the resultant observable
+ * is shared across all subscribers.
  */
- export function getBookEntities() {
-   return compose(fromBooks.getBookEntities(), getBooksState());
- }
-
- export function getBook(id: string) {
-   return compose(fromBooks.getBook(id), getBooksState());
- }
-
- export function hasBook(id: string) {
-   return compose(fromBooks.hasBook(id), getBooksState());
- }
-
- export function getBooks(bookIds: string[]) {
-   return compose(fromBooks.getBooks(bookIds), getBooksState());
- }
+ export const getBookEntities = share(compose(fromBooks.getBookEntities, getBooksState));
```

Comment:
`share()` might not make it into the final cut of the example app. If it does, it can be composed into a new function:
```ts
const shareAndCompose = compose(share, compose);
```